src/main/java/seedu/rc4hdb/model/person/AttributesMatchKeywordsPredicate.java

```diff
@@ -0,0 +1,48 @@
+package seedu.rc4hdb.model.person;
+
+import static java.util.Objects.isNull;
+
+import java.util.function.Predicate;
+
+import seedu.rc4hdb.logic.commands.modelcommands.FilterCommand;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class AttributesMatchKeywordsPredicate implements Predicate<Person> {
```

Comment:
Parts of this class could be further abstracted, let's split this up into smaller predicates, which will make up this larger predicate.